### ISSUE TITLE
Add language support to represent the current date and time

### ISSUE THREAD

**Issue by thiagodp** [thiagodp/concordialang#40]
Currently, Concordia relies on queries to get the current date, time or datetime. Examples:
```
- value comes from "SELECT CURRENT_DATE()"

- value comes from "SELECT CURRENT_TIME()"

- value comes from "SELECT NOW()"
```

Formatted values are also accepted:
```
# dd/mm/yyyy
- value comes from 'SELECT DATE_FORMAT( CURRENT_DATE(), "%d/%m/%Y")"

# hh:mm
- value comes from 'SELECT TIME_FORMAT( CURRENT_TIME(), "%h:%i")'

# dd/mm/yyyy hh:mm:ss
- value comes from 'SELECT DATE_FORMAT( NOW(), "%d/%m/%Y %h:%i:%s")'
```

*Instead, it should support a declaration that produces the value without having to deal with queries.*

### 1. Language constructions for date and time

The proposed constructions should consider the property `data type`. 
For example, `now` for a `date` must return the current date, while `now` for a `time` must return the current time.

### Date constructions
- `now`
- `current date`
- `today`
- `yesterday`
- `tomorrow`
- `next` + ( `week` | `month` | `year` )
  *Example:* `next week`
- `last` + ( `week` | `month` | `year` )
  *Example:* `last month`
- number + ( `day` | `days` | `week` | `weeks` | `year` | `years` ) + ( `ago` | `ahead` | `later` )
  *Example:* `2 days ago`

### Time constructions
- `now`
- `current time`
- `current` +  ( `second` | `minute` | `hour`  )
- number + ( `second` | `seconds` | `minute` | `minutes` | `hour` | `hours` ) + ( `ago` | `ahead` | `later` )
    *Example:* `1 second ago`

### Datetime constructions
- `current datetime`
- all the date and time constructions

### 2. Allow constructions for integer values

Example:

```concordia
UI Element: Sales Last Year
  - data type is integer
  - max value is the current year # current year value, like 2019

UI Element: Meeting Day
  - data type is integer
  - max value is today  # current day value, like 25
```
### Some Examples

```concordia
UI Element: Meeting Date
  - data type is date
  - min value is today

UI Element: Sales Last Year
  - data type is integer
  - max value is this year # current year value

UI Element: Birth Date
  - data type is date
  - max value is 18 years ago # the current date, but for 18 years ago
    Otherwise I see "You must be at least 18 years old."
```

### Future work

👉 New Issues may address them.

#### 1. Built-in functions `date()`, `time()`, and `datetime()` 

  - Both can receive an optional parameter `format`. Its value should default to the UI Element locale's date/time/datetime format, if available in the language (see #30). Otherwise, the locale defined by the document's language.
  - The parameter `format` should use the [PHP format for date and time values](https://www.php.net/manual/en/function.date.php).

  - By default, `date()` should return *day*, *month* and *year*. Example for the English locale: `12/31/2018`.

  - By default, `time()` should return *hour* and *minute* (without *second*). Example for the Portuguese locale: `13:00`.

  - By default, `datetime()` should return *day*, *month*, *year*, *hour*, *minute* and *second*. Example for the English locale: `12/31/2018 10:20:30`.

#### 3. Allow time zones (*as an additional optional parameter for the functions?*).

**Comment by thiagodp** [comment]
This feature is being released with [**Concordia 2**](https://github.com/thiagodp/concordialang/issues/56).

Constructions for integer values were changed. For `date` or `datetime`, you can use `day of`, `month of` or `year of` plus a date expression. Example: `- minimum value is year of last year`. For `time` or `datetime`, you can use `hour of`, `minute of` or `second of` plus a time expression. Example: `- maximum value is hour of 2 hours ago`.

Built-in date and time functions will be implemented in the future, as suggested.

**Comment by thiagodp** [comment]
Released in 2.0-alpha
